remote.go

```diff
@@ -292,30 +323,46 @@ func findLayerWithSha(layers []v1.Layer, sha string) (v1.Layer, error) {
 	return nil, fmt.Errorf(`previous image did not have layer with sha '%s'`, sha)
 }
 
-func (r *remoteImage) Save() (string, error) {
-	ref, auth, err := referenceForRepoName(r.keychain, r.repoName)
+func (r *RemoteImage) Save(additionalNames ...string) SaveResult {
```

Comment:
I would like to be able to still return an error type here for the sake of idiom. Could we make a custom error type instead of a custom result type to hold structured data (see ggcr transport.Error for example)?

Reply:
We opted for this solution due to the following.

1. We wanted to list out all "attempted" names in the results including the digest
2. Given that it could end in a partial success state we would end up with both "results" (digest and list of successful names) and errors which seemed to violate Golangs idiom. Plus GoLand was not happy.

Hope this makes sense but willing to discuss it further.

Reply:
Given that we have a separate function to return the digest could we use that instead to fetch the digest after saving (we would probably need to verify that it worked correctly in saved local image case)? That way we could avoid the error + results problem. We could make a custom error type that can provide errors per digest.